File: MPPCS/Service/SaleService.cs
using CommonDomain;
using Persistence.Repository;
using Persistence.Service.Validator;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Service
{
    public class SaleService
    {
        private SaleRepository repository;
        private SaleValidator validator;

        public SaleService(SaleRepository repository)
        {
            this.repository = repository;
            this.validator = new SaleValidator();
        }

        public void AddSale(string buyerName, int ticketsBought, int showId)
        {
            Sale sale = new Sale(buyerName, ticketsBought, showId);
            validator.validate(sale);
            repository.Add(sale);
        }
    }
}

File: PersistenceAndServices/src/main/java/service_base/SaleService.java
package service_base;

import domain.Sale;
import repository.SaleDatabaseRepository;
import service_base.validators.SaleValidator;

public class SaleService {
    private SaleDatabaseRepository repository;
    private final SaleValidator validator;

    public SaleService(SaleDatabaseRepository repository) {
        this.repository = repository;
        this.validator = new SaleValidator();
    }

    public void addSale (String buyerName, int ticketsBought, int showId) throws Exception {
        Sale sale = new Sale(buyerName,ticketsBought,showId);
        validator.validate(sale);
        repository.add(sale);
    }

    public Sale deleteSale(int id) throws Exception {
        repository.find(id);
        return repository.delete(id);
    }
}
